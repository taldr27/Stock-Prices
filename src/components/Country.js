import React from 'react';
import { NavLink } from 'react-router-dom';
import { useSelector } from 'react-redux';
import '../styles/Country.css';
import world from '../images/wrld.png';

const Country = () => {
  const countries = useSelector((state) => state.countries);
  return (
    <>
      <div className="top">
        <div style={{ minWidth: '100%' }}>
          <img
            src={world}
            alt="Example"
            className="example-img"
          />
          <p style={{ fontSize: '1.5rem' }}>World Companies</p>

        </div>
      </div>
      <div className="country-container">
        {countries.map((country) => (
          <div key={country.symbol}>
            <NavLink
              to={`/companies/${country.symbol}`}
            >
              <div id={country.symbol} className="img-container">
                <img
                  className="flagImg"
                  src={country.countryImg}
                  alt={country.countryName}
                />
                <p>{country.countryName}</p>
              </div>
            </NavLink>
          </div>
        ))}
      </div>
    </>
  );
};
export default Country;
