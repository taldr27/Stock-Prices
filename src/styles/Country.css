.top {
  display: flex;
  align-items: center;
  padding: 0.1em;
  background-color: hsl(219, 72%, 62%);
  color: white;
}

.country-container {
  display: grid;
  grid-template-columns: auto auto;
  justify-items: center;
  margin-inline: 0;
}

.example-img {
  width: 50%;
}

.country-container > * {
  background-color: hsl(219, 72%, 62%);
  border: 1px solid rgba(0, 0, 0, 0.172);
  font-size: 1.5rem;
}

.country-container > *:hover {
  background-color: hsl(219, 44%, 45%);
}

.img-container {
  color: white;
}

.flagImg {
  width: 100%;
}
