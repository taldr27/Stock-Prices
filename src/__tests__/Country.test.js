import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { StrictMode } from 'react';
import { Provider } from 'react-redux';
import Country from '../components/Country';
import store from '../redux/configureStore';

describe('Testing Country component', () => {
  it('Should match the snapshot Country', () => {
    const country = render(
      <StrictMode>
        <Router>
          <Provider store={store}>
            <Country />
          </Provider>
        </Router>
      </StrictMode>,
    );
    expect(country).toMatchSnapshot();
  });
});
