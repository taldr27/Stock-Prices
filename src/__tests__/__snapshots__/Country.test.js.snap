// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Country component Should match the snapshot Country 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="top"
      >
        <div
          style="min-width: 100%;"
        >
          <img
            alt="Example"
            class="example-img"
            src="wrld.png"
          />
          <p
            style="font-size: 1.5rem;"
          >
            World Companies
          </p>
        </div>
      </div>
      <div
        class="country-container"
      >
        <div>
          <a
            class=""
            href="/companies/US"
          >
            <div
              class="img-container"
              id="US"
            >
              <img
                alt="United States"
                class="flagImg"
                src="us.png"
              />
              <p>
                United States
              </p>
            </div>
          </a>
        </div>
        <div>
          <a
            class=""
            href="/companies/BR"
          >
            <div
              class="img-container"
              id="BR"
            >
              <img
                alt="Brasil"
                class="flagImg"
                src="br.png"
              />
              <p>
                Brasil
              </p>
            </div>
          </a>
        </div>
        <div>
          <a
            class=""
            href="/companies/MX"
          >
            <div
              class="img-container"
              id="MX"
            >
              <img
                alt="Mexico"
                class="flagImg"
                src="mx.png"
              />
              <p>
                Mexico
              </p>
            </div>
          </a>
        </div>
        <div>
          <a
            class=""
            href="/companies/GB"
          >
            <div
              class="img-container"
              id="GB"
            >
              <img
                alt="United Kingdom"
                class="flagImg"
                src="gb.png"
              />
              <p>
                United Kingdom
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="top"
    >
      <div
        style="min-width: 100%;"
      >
        <img
          alt="Example"
          class="example-img"
          src="wrld.png"
        />
        <p
          style="font-size: 1.5rem;"
        >
          World Companies
        </p>
      </div>
    </div>
    <div
      class="country-container"
    >
      <div>
        <a
          class=""
          href="/companies/US"
        >
          <div
            class="img-container"
            id="US"
          >
            <img
              alt="United States"
              class="flagImg"
              src="us.png"
            />
            <p>
              United States
            </p>
          </div>
        </a>
      </div>
      <div>
        <a
          class=""
          href="/companies/BR"
        >
          <div
            class="img-container"
            id="BR"
          >
            <img
              alt="Brasil"
              class="flagImg"
              src="br.png"
            />
            <p>
              Brasil
            </p>
          </div>
        </a>
      </div>
      <div>
        <a
          class=""
          href="/companies/MX"
        >
          <div
            class="img-container"
            id="MX"
          >
            <img
              alt="Mexico"
              class="flagImg"
              src="mx.png"
            />
            <p>
              Mexico
            </p>
          </div>
        </a>
      </div>
      <div>
        <a
          class=""
          href="/companies/GB"
        >
          <div
            class="img-container"
            id="GB"
          >
            <img
              alt="United Kingdom"
              class="flagImg"
              src="gb.png"
            />
            <p>
              United Kingdom
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
